src/components/product.js

```diff
@@ -52,6 +52,22 @@ function whitelistedProperties(selectorStyles) {
   }, {});
 }
 
+function adaptConfig(config) {
+  if (config.id && !isBase64(config.id)) {
+    config.id = btoa(`gid://shopify/Product/${config.id}`);
+  }
+
+  return config;
+}
+
+function isBase64(str) {
+  try {
+    return btoa(atob(str)) === str;
```

Comment:
I'm not sure this is the best way to do this.

```javascript
str = '1234'

btoa(atob(str)) === str
```

The above resolves to true.

It may make sense to have an `id` field, which is deprecated, and a `storefrontId` field, which is the new ID format.

The base64 encoding format doesn't make this a deterministic check if the string is a multiple of 4 characters long.